### PR TITLE
Introduce deletion-mode for assets

### PR DESCRIPTION
**Pull request by nicoloboschi**
Same of https://github.com/LangStream/langstream/pull/418 but for assets.

new field `deletion-mode: none|delete`. Default is none

```
assets:
- name: "vsearch-keyspace"
    asset-type: "cassandra-keyspace"
    creation-mode: create-if-not-exists
    deletion-mode: delete
    config:
      keyspace: "vsearch"
      datasource: "CassandraDatasource"
      create-statements:
        - "CREATE KEYSPACE vsearch WITH REPLICATION = {'class' : 'SimpleStrategy','replication_factor' : 1};"
```

### CODE REVIEW

File: langstream-agents/langstream-vector-agents/src/main/java/ai/langstream/agents/vector/cassandra/CassandraAssetsManagerProvider.java [hunk 1]
```diff
@@ -111,18 +117,31 @@ public void deployAsset() throws Exception {
...
+        }
+
         @Override
-        public void deleteAsset() throws Exception {
-            throw new UnsupportedOperationException();
+        public boolean deleteAssetIfExists() throws Exception {
+            final String keySpace = getKeySpace();
+            List<String> statements =
+                    ConfigurationUtils.getList("delete-statements", assetDefinition.getConfig());
+            if (statements.isEmpty()) {
```
Comment (eolivelli) [inline langstream-agents/langstream-vector-agents/src/main/java/ai/langstream/agents/vector/cassandra/CassandraAssetsManagerProvider.java:131]: I don't think this is a good idea, the table may have other dependencies like INDEXes it really depends on how you created it (and we have no control over the create-statements)

if there is no statement then we do nothing, like for create-statements

File: langstream-agents/langstream-vector-agents/src/main/java/ai/langstream/agents/vector/cassandra/CassandraAssetsManagerProvider.java [hunk 2]
```diff
@@ -154,8 +182,14 @@ public void deployAsset() throws Exception {
         }
 
         @Override
-        public void deleteAsset() throws Exception {
-            throw new UnsupportedOperationException();
+        public boolean deleteAssetIfExists() throws Exception {
+            List<String> statements =
+                    ConfigurationUtils.getList("delete-statements", assetDefinition.getConfig());
+            if (statements.isEmpty()) {
+                statements = List.of("DROP KEYSPACE IF EXISTS %s;".formatted(getKeyspace()));
```
Comment (eolivelli) [inline langstream-agents/langstream-vector-agents/src/main/java/ai/langstream/agents/vector/cassandra/CassandraAssetsManagerProvider.java:189]: also here

**Review by eolivelli — CHANGES_REQUESTED** [review]

File: langstream-agents/langstream-vector-agents/src/main/java/ai/langstream/agents/vector/cassandra/CassandraAssetsManagerProvider.java (hunk 1 above)
Comment (nicoloboschi) [reply langstream-agents/langstream-vector-agents/src/main/java/ai/langstream/agents/vector/cassandra/CassandraAssetsManagerProvider.java:131]: it's not very intuitive. 
You set deletion-mode: delete and then you have to provide the keyspace delete statement ?
It's much easier if you just delete it. 

From my testing, the keyspace is being deleted even if not empty and it works well. 

Do you think an INDEX would make the keyspace deletion fail ?

**Review by eolivelli — APPROVED** [review]
if the auto delete command works with tables with  indexed (for instance of Vector columns) then I am +1


as an alternative we can fail the deployment here

https://github.com/LangStream/langstream/blob/965c78e7261471743c70dc13cf9e5d9ad316b3e6/langstream-core/src/main/java/ai/langstream/impl/assets/CassandraAssetsProvider.java#L30

in any case we must add the validation in CassandraAssetsProvider

**Pull request merged**